### DISCUSSION TITLE
Welcome to ConfuserEx discussions!

### DISCUSSION THREAD

**Discussion by mkaring** [mkaring/ConfuserEx#318] [General]
## 👋 Welcome!

Let's use Discussions as a place ask questions, collaborate on new ideas, and connect with members of our community. Feel free to:

* Ask any questions you’re wondering about
* Share ideas
* Engage with other community members

Please be sure to welcome others and are open-minded.

**Comment by NaveenBadal** [comment]
Hello,

I built a custom application , in which I integrated the ConfuserEx.CLI application. I am running the ConfuserEx.CLI on background and showing the output in my custom textbox that I am receiving from command line. 

My first question is that:
Is there any way to get the progress bar value from command line, as when the enabling newtonsoft.json compatibility step is running, it is almost taking half an hour and may be more. So I want to show a progress in that.

Second question is that:
Why is NewtonSoft.Json Enabling Compatibility step talking too much time? Is there any way to reduce its time?

**Reply by mkaring** [reply]
Hi,

Currently there is no way to get any report of the progress from the ConfuserEx CLI.

The Newtonsoft.Json compatiblity message isn't taking any time. It's just the last log message, before ConfuserEx is starting to analyze the structure and cross references of the entire assembly. Depending on the size and complexity of the assemblies you are working with, this may take some time. Mind you: Half an hour seems very excessive. What size of project are we talking about here?

**Reply by NaveenBadal** [reply]
Hello,

The count of DLLs that I am obfuscating is around 35. But not all of the DLLs uses the NewtonSoft.json. Only some of them uses it. I tried with both Confuser Ex GUI tool and with ConfuserEx.CLI, in both the tool, all other processes are not taking time, but this Found Newtonsoft.json, enabling compatibilty alone taking lot of time. On an average it is taking around 30 minutes. Maximum it takes 45 minutes to perform this step. Yes the project is big, but I don't think so, It should take that much time. Because all others steps, together will complete may be in 5 to 10 minutes.

**Reply by mkaring** [reply]
The `Newtonsoft.JSON` compatibility is enabled, if it's detected in _any_ assembly. How ever as I said: Once that message appears the compatibility is already enabled and  the work this step does is done.

It's impossible to tell what is taking so long in the analysis step for the rename protection that causes you so see such an extensive delay, without the assemblies that are causing this issue.

**Comment by AdamShakhabov** [comment]
### What value of "rule.preset" attribute does mean?

I am not sure that completely understood how `minimum`, `normal`, [etc. values](https://github.com/mkaring/ConfuserEx/blob/master/docs/ProjectFormat.md#element-rule) of `rule.preset` affect on obfuscation. Each value contains predefined set of protections — "anti debug", "anti dump", "anti ildasm" etc?

![image](https://user-images.githubusercontent.com/5969304/134958111-5818e591-d5f6-4875-b3cf-dfb472c004ce.png)

**Reply by mkaring** [reply]
The presets are, as you already suspected, a predefined set of protections that are applied if you select that option.
You'll find in the wiki what protection is enabled at what preset. Protections that are part of a lower preset, are also enabled on a higher preset.

For example: The [`anti debug`](https://github.com/mkaring/ConfuserEx/wiki/Anti-Debug-Protection) protection is part of the "Minimal" preset. Meaning if you choose, "Minimal", "Normal", "Aggressive" or "Maximum" it will be enabled.
The [`anti tamper`](https://github.com/mkaring/ConfuserEx/wiki/Anti-Tamper-Protection) protection is part of the "Maximum" preset and will only be enabled when choosing "Maximum" as a preset.

**Comment by chayush24** [comment]
Hi Team,

I want to use ConfuserEx to protect (obfuscate/encrypt) a **.NET 8 DLL**.

From the repository README, I can see support mentioned for:
- .NET Framework (2.0 – 4.8)

But I could not find **clear confirmation for .NET 8 / modern .NET (Core / 5 / 6 / 7 / 8)**.

### My questions:
1. Does ConfuserEx officially support **.NET 8** assemblies?
2. Has anyone successfully used ConfuserEx with **.NET 8 DLLs** in production?
3. If not supported, are there plans or branches (like Reborn) that work with .NET 8?
4. Are there any known issues related to metadata/runtime changes in .NET 8?

I want to use it for:
- DLL obfuscation
- Code encryption
- Tamper protection

Any clarification from maintainers or experienced users would be really helpful.

Thanks in advance 🙏
